services/datamanager/builtin/builtin.go

```diff
@@ -217,73 +234,66 @@ func getDurationFromHz(captureFrequencyHz float32) time.Duration {
 // Initialize a collector for the component/method or update it if it has previously been created.
 // Return the component/method metadata which is used as a key in the collectors map.
 func (svc *builtIn) initializeOrUpdateCollector(
-	attributes dataCaptureConfig) (
-	*componentMethodMetadata, error,
+	md componentMethodMetadata,
+	config dataCaptureConfig) (
+	*collectorAndConfig, error,
 ) {
-	// Create component/method metadata to check if the collector exists.
-	metadata := data.MethodMetadata{
-		Subtype:    attributes.Type,
-		MethodName: attributes.Method,
-	}
-
-	componentMetadata := componentMethodMetadata{
-		ComponentName:  attributes.Name,
-		ComponentModel: attributes.Model,
-		MethodParams:   fmt.Sprintf("%v", attributes.AdditionalParams),
-		MethodMetadata: metadata,
-	}
 	// Build metadata.
-	captureMetadata, err := datacapture.BuildCaptureMetadata(attributes.Type, attributes.Name,
-		attributes.Model, attributes.Method, attributes.AdditionalParams, attributes.Tags)
+	captureMetadata, err := datacapture.BuildCaptureMetadata(config.Type, config.Name,
```

Comment:
Thanks for the rename here, as we might start using the config's component Attributes soon which would have gotten confusing.